### PR TITLE
Project Solana Address Support

### PR DESCRIPTION
**Pull request by aminlatifi**
#1187

### CODE REVIEW

File: src/resolvers/projectResolver.ts [hunk 1]
```diff
@@ -475,140 +482,84 @@ export class ProjectResolver {
...
+          return;
+        case FilterField.AcceptFundOnSolana:
+          acceptFundOnSolanaSeen = true;
+          return;
+        default:
+          return query.andWhere(`project.${filter} = true`);
+      }
+    });
+
+    if (networkIds.length > 0 || acceptFundOnSolanaSeen) {
```
Comment (CarlosQ96) [inline src/resolvers/projectResolver.ts:555]: We will have to limit it in the frontend. Like you said both are not posible

**Review by CarlosQ96 — APPROVED** [review]
Looks great functionality wise, I will add the tokens in my feature branch for the donation.

File: src/resolvers/projectResolver.ts (hunk 1 above)
Comment (aminlatifi) [reply src/resolvers/projectResolver.ts:555]: No, I believe we don't need to limit the front end. Before this, we had an `AND` conjunction between accepting funds on chains, so choosing multiple ones had no meaning. After this, we will have an `OR` conjunction between them.

File: src/resolvers/projectResolver.test.ts [hunk 2]
```diff
@@ -1262,6 +1277,145 @@ function allProjectsTestCases() {
...
+      chainType: ChainType.SOLANA,
+      networkId: 0,
+      isRecipient: true,
+    });
+    await solanaAddress.save();
+
+    const result = await axios.post(graphqlUrl, {
+      query: fetchMultiFilterAllProjectsQuery,
+      variables: {
+        filters: ['AcceptFundOnMainnet', 'AcceptFundOnSolana'],
```
Comment (aminlatifi) [inline src/resolvers/projectResolver.test.ts:1499]: @CarlosQ96 Look at this test, all other test scenarios we have had before only test filters with single critrion.

**Review by mohammadranjbarz — APPROVED** [review]
Thanks @aminlatifi  for implementing Solana and refactoring some old codes.
LGTM, but I think it's better to add `chainType` to admin bro then admins can filter donations/tokens based on chainType

https://github.com/Giveth/impact-graph/blob/248356696da9a949101d3dd4e911b33159c97154/src/server/adminJs/tabs/tokenTab.ts#L174

https://github.com/Giveth/impact-graph/blob/248356696da9a949101d3dd4e911b33159c97154/src/server/adminJs/tabs/donationTab.ts#L372

**Comment by aminlatifi** [comment]
Closed and made #1200

**Comment by aminlatifi** [comment]
> Thanks @aminlatifi for implementing Solana and refactoring some old codes. LGTM, but I think it's better to add `chainType` to admin bro then admins can filter donations/tokens based on chainType
> 
> https://github.com/Giveth/impact-graph/blob/248356696da9a949101d3dd4e911b33159c97154/src/server/adminJs/tabs/tokenTab.ts#L174
> 
> https://github.com/Giveth/impact-graph/blob/248356696da9a949101d3dd4e911b33159c97154/src/server/adminJs/tabs/donationTab.ts#L372

Let's solve it as a separate issue, because it's not clear for me how it would be!

**Pull request closed without merging**